src/lib.rs

```diff
@@ -1,268 +1,211 @@
 
-# let mut app = App::new();
-# app.add_plugins(EnhancedInputPlugin);
-app.add_input_context::<Player>();
+Inside of this crate, actions are stored as entities with the [`Action<A>`] component, where `A` is your [`InputAction`] type.
+These are associated to contexts via the [`ActionOf<C>`] relationship, where `C` is your context type,
+and can be quickly bound to them using the [`actions!`] macro.
 
-# let mut world = World::new();
-world.spawn((
-    Player,
-    actions!(Player[
-        (
-            Action::<Jump>::new(),
-            bindings![KeyCode::Space, GamepadButton::South],
-        ),
-        (
-            Action::<Fire>::new(),
-            bindings![MouseButton::Left, GamepadButton::RightTrigger2],
-        ),
-    ])
-));
+The behavior of each action can be further customized using the [`ActionSettings`] component.
 
-#[derive(Component)]
-struct Player;
+## Bindings
 
-#[derive(InputAction)]
-#[action_output(bool)]
-struct Jump;
+Bindings define how actions are triggered by input sources (e.g. mouse movement or keyboard buttons) that your player might press, like keyboard keys or gamepad buttons.
+We provide support for a variety of input sources out of the box: see the [`Binding`] enum for a full list.
 
-#[derive(InputAction)]
-#[action_output(bool)]
-struct Fire;
-```
+Inside of this crate, bindings are represented by entities with the [`Binding`] component.
+Bindings associated with actions via [`BindingOf`] relationship. Similar to [`actions!`],
+we provide the [`bindings!`] macro to spawn related bindings.
 
 By default, input is read from all connected gamepads. You can customize this by adding the [`GamepadDevice`] component to the
 context entity.
 
-Context actions will be evaluated in the schedule associated at context registration. Contexts registered in the same
-schedule will be evaluated in their spawning order, but you can override it by adding the [`ContextPriority`] component.
-You can also activate or deactivate contexts by inserting [`ContextActivity`] component.
+## Contexts
+
+Contexts define when actions are evaluated. They are associated with action entities via the [`Actions<C>`] relationship mentioned earlier.
+Depending on your type of game, you may have a single global context
+or multiple contexts for different gameplay states. For games with multiple entities driven by a single context it's
+common to create a "controller" entity which applies the input to the desired entity.
+
+Inside of this crate, contexts are stored using regular components, commonly on an entity for which the input is associated (player, button, dialog, etc.).
+
+Contexts can be activated or deactivated using the [`ContextActivity`] component.
+By default, contexts are active when the component is present.
+When active, all actions associated with the context are evaluated.
+
+Contexts can be [layered](crate::context::ContextPriority), defining a priority order, and any number of them can be active at the same time.
```

Comment:
We previously mentioned that multiple contexts can be active at the same time. Maybe write it like this?
```suggestion
By default contexts evaluated in their spawn order, but you can also control this with [`ContextPriority`].
```